File: frontend/src/app/models/professor.ts
import { Funcionario } from './funcionario';
import { Usuario } from './usuario';
import { Turma } from './turmas';
import { Disciplina } from './disciplina';

export class Professor {
  id?: number;
  funcionario: Funcionario | null;
  usuario: Usuario | null;
  turmas: Turma[];
  disciplinas: Disciplina[];

  constructor(
    funcionario: Funcionario | null,
    usuario: Usuario | null,
    turmas: Turma[] = [],
    disciplinas: Disciplina[] = [],
    id?: number
  ) {
    this.id = id;
    this.funcionario = funcionario;
    this.usuario = usuario;
    this.turmas = turmas;
    this.disciplinas = disciplinas;
  }
}

File: backend/src/main/java/com/sentinel/backend/entity/Professor.java
package com.sentinel.backend.entity;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "professores")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Professor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "funcionario_id")
    private Funcionario funcionario;

    @OneToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @ManyToMany
    @JoinTable(name = "professor_turmas", joinColumns = @JoinColumn(name = "professor_id"), inverseJoinColumns = @JoinColumn(name = "turma_id"))
    private List<Turma> turmas;

    @ManyToMany
    @JoinTable(name = "professor_disciplinas", joinColumns = @JoinColumn(name = "professor_id"), inverseJoinColumns = @JoinColumn(name = "disciplina_id"))
    private List<Disciplina> disciplinas;
}
